### ISSUE TITLE
Block usage of `p`

### ISSUE THREAD

**Issue by grosser** [rubocop/rubocop#12824]
## Is your feature request related to a problem? Please describe.

I often leave debugging code by accident like:
```
foo = p(bar)
```

## Describe the solution you'd like

Warn me about it + auto-remove

## Describe alternatives you've considered

Being more careful

**Comment by koic** [comment]
`p` is excluded from the default due to false positive reports:
https://github.com/rubocop/rubocop/pull/11564

It cannot be made a default, but users can extend it in your `.rubocop.yml`:

```yaml
Lint/Debugger:
  DebuggerMethods:
    Kernel:
      - p
```

**Comment by grosser** [comment]
yeah makes sense, sadly setting this overrides everything else and so I got
... but I assume making these merge instead of override would be a breaking change too 😞 

```
Rakefile:71:17: W: [Corrected] Lint/RedundantCopDisableDirective: Unnecessary disabling of Lint/Debugger.
    binding.irb # rubocop:disable Lint/Debugger
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```
